scripts/zephyr_module.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2019, Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''Tool for parsing a list of projects to determine if they are Zephyr
+projects. If no projects are given then the output from `west list` will be
+used as project list.
+
+Include files can be generated for Kconfig and/or CMake using --kconfig-out and
+--cmake-out arguments
+'''
+
+import argparse
+import os
+import sys
+import yaml
+import pykwalify.core
+import subprocess
+import re
+
+
+METADATA_SCHEMA = '''\
+## A pykwalify schema for basic validation of the structure of a
+## metadata YAML file.
+##
+# The zephyr/module.yml file is a simple list of key value pairs to be used by
+# the build system.
+type: map
+mapping:
+  build:
+    required: true
+    type: map
+    mapping:
+      cmake:
+        required: false
+        type: str
+      kconfig:
+        required: false
+        type: str
+'''
+
+AUTO_GENERATED_HEADER = '# NOTE: THIS FILE IS AUTOGENERATED - DO NOT EDIT\n\n'
+
+
+def process_module(module, section_name, cmake_out=None, kconfig_out=None):
+    cmake_path = 'zephyr'
+    kconfig_file = 'zephyr/Kconfig'
+
+    module_yml = os.path.join(module, 'zephyr/module.yml')
+    if os.path.isfile(module_yml):
+        with open(module_yml, 'r') as f:
+            meta = yaml.safe_load(f.read())
+
+        schema = yaml.safe_load(METADATA_SCHEMA)
```

Comment:
You might need a `Loader=` param to avoid warnings with newer versions of PyYaml: https://github.com/zephyrproject-rtos/ci-tools/commit/a5298663078163a818fcf7d2c04e379050b56057

Reply:
Thanks, will add.

Reply:
Took a look at: zephyrproject-rtos/ci-tools@a529866
That code uses `yaml.load` and not `yaml.safe_load` as this code.

So that is not an issue here (as well as safe_load doesn't support the `Loader` argument)

Reply:
It also feels a bit strange to parse this schema for each module, since it isn't changing. It could either be a global variable or passed as an argument instead to avoid that.

Reply:
@mbolivar fixed